graspologic/embed/lse.py

```diff
@@ -161,4 +235,10 @@ def fit(self, graph, y=None):
         self.n_features_in_ = A.shape[0]
         L_norm = to_laplacian(A, form=self.form, regularizer=self.regularizer)
         self._reduce_dim(L_norm)
+
+        # for out-of-sample
+        inv_eigs = np.diag(1 / self.singular_values_)
+        self._pinv_left = self.latent_left_ @ inv_eigs
+        if self.latent_right_ is not None:
+            self._pinv_right = self.latent_right_ @ inv_eigs
```

Comment:
Since this code is so similar to the ASE oos code, I'm wondering if we can just refactor into the superclass to avoid code repitition?

Reply:
I moved transform to the BaseSpectralEmbed, but the fit methods are slightly different. Do we think it would be better to move the code that is the same in lse and ase fit to BaseSpectralEmbed, or would it be better to repeat a little bit of code?

Reply:
imo, it's almost always better to avoid repeating code if possible, so i'd see if you can find a way to move the code that is the same up to the base class

Reply:
@loftusa is there a reason why you put this at the end of fit rather than the beginning of transform? Seems like it's not a necessary calculation if someone doesn't have oos nodes.

Reply:
mainly because it's an O(not much) set of calculations and I figured it might at some point be useful to have access to `self_pinv_left` prior to the transform, but I'm not at all opposed to changing it if you think it makes more sense

Reply:
that way we can put this all into transform instead of having the extra method in base. Also think it'll look cleaner.

Reply:
Sounds good, ill move this all to transform